dev/Repeater/APITests/InspectingDataSourceTests.cs

```diff
@@ -115,6 +115,22 @@ public void VerifyUniqueIdMappingInterface()
             });
         }
 
+        [TestMethod]
+        public void VerifyIndexOfBehavior()
+        {
+            RunOnUIThread.Execute(() =>
+            {
+                var data = new ObservableVectorWithUniqueIds(Enumerable.Range(0, 10));
```

Comment:
can you please also add tests for Enumerable and  ObservableCollection

Reply:
Added them. Using just an IEnumerable, calling IndexOf throws an exception, that is expected right (as casting to IVector does not work)?

Reply:
Hmm.. weird, can you debug that case? I think [IEnumerable maps to IIterable](https://docs.microsoft.com/en-us/uwp/api/Windows.Foundation.Collections.IIterable_T_), so it should go through WrapIterable call which will end up creating a vector and that should work...

Reply:
It seems that Enumerable.Range() is the problem, as it generates an IEnumerable<int>. That seems to be problematic as int will be passed by value, not by reference.
Is this something we need to take care of in the API (by allowing ref's for example) ?